app/assets/javascripts/models/beta/story.js

```diff
@@ -330,27 +330,38 @@ export const mergeWithFetchedStories = (
   fetchedStories,
   pastStoryIds
 ) => {
-  const storyNotSaved = currentStories.filter(isNew);
-
-  const editedStories = fetchedStories.map((fetchedStory) => {
-    const existingStory = currentStories.find(
-      (story) => story.id === fetchedStory.id
-    );
-    if (existingStory && !existingStory.collapsed) {
-      return {
-        ...fetchedStory,
+  const mergedStories = { ...fetchedStories };
+
+  if (Object.values(currentStories).length === 0) {
+    return mergedStories;
+  }
+
+  if (currentStories.stories[null]) {
+    mergedStories.stories[null] = currentStories.stories[null];
+  }
+
+  Object.values(fetchedStories.stories).map((fetchedStory) => {
+    const storyId = fetchedStory.id;
+
+    if (
+      currentStories.stories[storyId] &&
+      !currentStories.stories[storyId].collapsed
+    ) {
+      mergedStories.stories[storyId] = {
+        ...fetchedStories.stories[storyId],
         collapsed: false,
-        _editing: existingStory._editing,
+        _editing: currentStories.stories[storyId]._editing,
       };
     }
-    return fetchedStory;
   });
 
-  const pastStories = pastStoryIds.map((id) =>
-    currentStories.find((story) => story.id === id)
-  ).filter((story) => story !== undefined);
+  pastStoryIds.forEach((storyId) => {
+    if (currentStories.stories[storyId]) {
+      mergedStories.stories[storyId] = currentStories.stories[storyId];
```

Comment:
This will also cause mutability. I'll give a hint on how to solve it: try to avoid `forEach` in this whole method and use `map` and `reduce` instead, without attribute assignment, just returning and spreading objects.

Reply:
Reviewing the code I realized that I don't need this data anymore. As i'm working with a collection of data, there's no need to keep adding the past stories to my collection again. Also, I no need the list of pastStoryIds, which greatly simplifies the code!

Now, I just need to say that ```mergedStories = { ...currentStories }``` and update just the fetchedStories in the currentStories.